torch/autograd/_functions/blas.py

```diff
@@ -17,8 +17,21 @@ class Addmm(InplaceFunction):
 
     @staticmethod
     def symbolic(g, add_matrix, matrix1, matrix2, beta=1, alpha=1, inplace=False):
-        # TODO: test if broadcasting occurred
-        return g.op("Gemm", matrix1, matrix2, add_matrix, beta_f=beta, alpha_f=alpha, broadcast_i=True)
+        assert matrix1.hasType() and matrix2.hasType() and add_matrix.hasType()
+        sizes1 = matrix1.type().sizes()
+        sizes2 = matrix2.type().sizes()
+        sizes_add = add_matrix.type().sizes()
+        broadcast = False
+        assert len(sizes1) + len(sizes2) - 2 >= len(sizes_add)
+        if (len(sizes1) + len(sizes2) - 2 > len(sizes_add)):
+            broadcast = True
+        else:
+            sizes12 = sizes1[:-1] + sizes2[1:]
```

Comment:
This is an interesting case: IIUC, Caffe2 wouldn't actually support broadcast in this case: I believe Caffe2 only supports broadcasting a 1D tensor into a 2D one, and not expanding a 3 x 1 tensor into a 3 x 9 one (for example). ONNX docs seem to imply that broadcasting is not supported in the latter case https://github.com/onnx/onnx/blob/master/docs/Operators.md#Add (or perhaps I misunderstand the logic here?)

Reply:
@apaszke Sure. I will move these new tests to onnx-pytorch.

Reply:
Okay, let me check with Junjie. The description of broadcasting in GEMM's doc is a little bit vague. It should be good, if we explicitly say what we support in ONNX, which we don't.

If we try to export some GEMM operator which is not directly supported by ONNX, probably we have to either raise the exception or change the shape by ourselves in the exporter.